### PR TITLE
Synchronization tweaks for IPC services

### LINKED ISSUE #2027: Parity won't die
```
2016-08-31 16:56:48 Unauthorized connection to Signer API blocked.
2016-08-31 16:56:50 Unauthorized connection to Signer API blocked.
^C2016-08-31 16:56:52 Finishing work, please wait...
2016-08-31 16:56:55     0/ 2/25 peers     40 MiB db    2 MiB chain    2 KiB queue   29 KiB sync
^C2016-08-31 16:57:16 Syncing #2089082 d411…dbf1      0 blk/s    0 tx/s   0 Mgas/s       0+    0 Qed   #2089082    1/ 3/25 peers     40 MiB db    3 MiB chain    2 KiB queue   29 KiB sync
2016-08-31 16:57:21 Syncing #2089082 d411…dbf1      0 blk/s    0 tx/s   0 Mgas/s       0+    0 Qed   #2089082    1/ 3/25 peers     40 MiB db    3 MiB chain    2 KiB queue   29 KiB sync
2016-08-31 16:57:26 Syncing #2089082 d411…dbf1      0 blk/s    0 tx/s   0 Mgas/s       0+    0 Qed   #2089081    1/ 3/25 peers     40 MiB db    3 MiB chain    2 KiB queue   29 KiB sync
2016-08-31 16:57:31 Syncing #2089082 d411…dbf1      0 blk/s    0 tx/s   0 Mgas/s       0+    0 Qed   #2089081    1/ 3/25 peers     40 MiB db    3 MiB chain    2 KiB queue   29 KiB sync
^C^C^C^C
```

```
$ ps aux | grep parity
megatv          25172  38.2  1.5  2857728 257232 s000  R+    4:55pm   0:41.06 parity
megatv          25173   6.7  0.1  2570288  22736 s000  S+    4:55pm   0:05.21 /usr/local/bin/parity sync
```

```
$ parity --version
Parity
  version Parity/v1.4.0-unstable-25e6a4e-20160831/x86_64-macos/rustc1.11.0
Copyright 2015, 2016 Ethcore (UK) Limited
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

By Wood/Paronyan/Kotewicz/Drwięga/Volf.
```

OS X

happened twice in a row after getting a new build

### PR DESCRIPTION
**Pull request by NikVolf**
fixes #2027

### CODE REVIEW

File: ipc/hypervisor/src/service.rs.in [hunk 1]
```diff
@@ -42,7 +42,7 @@ pub struct ModuleState {
 
 #[derive(Ipc)]
 pub trait ControlService {
-	fn shutdown(&self);
+	fn shutdown(&self) -> bool;
```
Comment (NikVolf) [inline ipc/hypervisor/src/service.rs.in]: Having a return type forces invocation to be synchronous
Until the exact issue will be found

**Comment by debris** [comment]
ctrl-c crashed the node with the following error:

```
2016-09-01 11:00:11     0/11/25 peers     41 MiB db    8 KiB chain    2 KiB queue   29 KiB sync
^C2016-09-01 11:00:39 Finishing work, please wait...
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Serialization(BinaryConvertError { member_tree: [], kind: SizeMismatch { expected: 1, found: 0 } })', ../src/libcore/result.rs:788
stack backtrace:
   1:        0x10e69889b - std::sys::backtrace::tracing::imp::write::h46e546df6e4e4fe6
   2:        0x10e69dd3a - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h077deeda8b799591
   3:        0x10e69d96a - std::panicking::default_hook::heb8b6fd640571a4f
   4:        0x10e682ed8 - std::panicking::rust_panic_with_hook::hd7b83626099d3416
   5:        0x10e69e316 - std::panicking::begin_panic::h941ea76fc945d925
   6:        0x10e683b18 - std::panicking::begin_panic_fmt::h30280d4dd3f149f5
   7:        0x10e69df6f - rust_begin_unwind
   8:        0x10e6c48f0 - core::panicking::panic_fmt::h2d3cc8234dde51b4
   9:        0x10e30bdf9 - core::result::unwrap_failed::h7fb1eff49b332d5c
  10:        0x10e30b1d9 - ethcore_ipc_hypervisor::service::HypervisorService::send_shutdown::hda75b4ec5c8d4a5f
  11:        0x10e30a0bc - ethcore_ipc_hypervisor::Hypervisor::shutdown::h3b511e62310e030e
  12:        0x10d9b1992 - parity::run::execute::h497d80819edf3ae9
  13:        0x10d99dd17 - parity::execute::h571dbc6454b59b6f
  14:        0x10da24f7c - parity::main::hc0b8b9a4c77213f8
  15:        0x10e69d52d - std::panicking::try::call::hca715a47aa047c49
  16:        0x10e69e3eb - __rust_try
  17:        0x10e69e385 - __rust_maybe_catch_panic
  18:        0x10e69d351 - std::rt::lang_start::h162055cb2e4b9fe7
```

**Comment by debris** [comment]
and after some time (looks like process was not closed)

```
--- ethcore/parity ‹ipc-control-svc-seqcst* ?› » thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Error { repr: Custom(Custom { kind: TimedOut, error: StringError("Operation timed out") }) })', ../src/libcore/result.rs:788
stack backtrace:
   1:        0x105eb2b1b - std::sys::backtrace::tracing::imp::write::h46e546df6e4e4fe6
   2:        0x105eb7f7a - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h077deeda8b799591
   3:        0x105eb7baa - std::panicking::default_hook::heb8b6fd640571a4f
   4:        0x105e9d178 - std::panicking::rust_panic_with_hook::hd7b83626099d3416
   5:        0x105eb8556 - std::panicking::begin_panic::h941ea76fc945d925
   6:        0x105e9ddb8 - std::panicking::begin_panic_fmt::h30280d4dd3f149f5
   7:        0x105eb81af - rust_begin_unwind
   8:        0x105edeb30 - core::panicking::panic_fmt::h2d3cc8234dde51b4
   9:        0x10539acde - core::result::unwrap_failed::h7fb1eff49b332d5c
  10:        0x10522eb38 - parity::sync::main::hf03be485bc162ca1
  11:        0x1052216ee - parity::main::hc0b8b9a4c77213f8
  12:        0x105eb776d - std::panicking::try::call::hca715a47aa047c49
  13:        0x105eb862b - __rust_try
  14:        0x105eb85c5 - __rust_maybe_catch_panic
  15:        0x105eb7591 - std::rt::lang_start::h162055cb2e4b9fe7
```

**Comment by NikVolf** [comment]
@debris probably you have orphanned process running

**Pull request merged**